extensions/amp-selector/0.1/amp-selector.js

```diff
@@ -89,10 +130,10 @@ export class AmpSelector extends AMP.BaseElement {
       this.clearAllSelections_();
       return;
     }
-    let selectedArray = Array.isArray(newValue) ? newValue : [newValue];
+    const selectedArray = Array.isArray(newValue) ? newValue : [newValue];
     // Only use first value if multiple selection is disabled.
     if (!this.isMultiple_) {
-      selectedArray = selectedArray.slice(0, 1);
+      selectedArray.length = 1;
```

Comment:
Ooof, sorry, but this introduced a bug itself. Imagine passing in `[]` as the newly selected values. Now, we change length to `1`, and we get `[undefined]` (sparse array implicit undefined). Sorry, but can you change this back? Or `Math.min(array.length, 1)`